HW04.Rmd

```diff
@@ -29,33 +29,91 @@ Knitr is part of the knitr package. Allows you to change column names and add ca
 
 
 ```{r month, warning=FALSE}
-
+number_per_month <- gun_deaths %>% 
+    select(year, month) %>% 
+    group_by(year) %>%
+    count(month) #select and count number of deaths per month
+
+number_per_month$month <- factor(number_per_month$month) 
+levels(number_per_month$month) <- c("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec") #convert month to abbreviation
+    knitr::kable(x = number_per_month, format = "markdown", col.names = c("Year", "Month", "Number of Deaths"), align = "ccc", caption = "Number of Gun Deaths Per Month (2012-2014)")
 ```
 
 
 ### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
 
 ```{r month_chart}
-
+number_per_month %>% 
+    ggplot(aes(x = month, y = n)) +
+    geom_col(fill = "#00CDFF") +
+    labs(title = "Number of Gun Deaths by Month (2012-2014)",
+         x = "Month", 
+         y = "Number of Deaths",
+         caption = "Source: FiveThirtyEight") +
+    theme_classic() +
+    theme(plot.title = element_text(hjust = 0.5))
+    
 ```
 
 ## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
 
 ```{r intent}
-
+gun_deaths %>% 
+    select(intent) %>% 
+    count(intent) %>% #select and count by intent
+    arrange(desc(n)) %>% #arrange by number of deaths in descending order
+    mutate(intent = factor(intent, levels = intent)) %>% #make sure that the bars are sorted in descending order of n
+    ggplot(aes(x = intent, y = n)) +
+      geom_col(fill = "#00CDFF") +
+    geom_text(aes(label = n), vjust = -0.5) +
+      labs(title = "Number of Gun Deaths by Intent",
+           x = "Intent",
+           y = "Number of Deaths", 
+           caption = "Source: FiveThirtyEight") +
+      theme_classic() +
+      theme(plot.title = element_text(hjust = 0.5))
 ```
 
 ## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
 
 ```{r age_&_sex}
+number_sex_age <- gun_deaths %>% 
+    select(sex, age) %>% 
+    filter(age != "Not Available") %>% 
+    type.convert() 
+levels(number_sex_age$sex) <- c("Female", "Male")
+#convert age to numeric and sex to factor
+#plot
+ggplot(number_sex_age, aes(x = sex, y = age)) +
+    geom_boxplot(outlier.alpha = 0, fill = "#00CDFF") +
+    labs(title = "Age Distribution of Gun Death Victims by Sex (2012-2014)",
+         x = "Sex",
+         y = "Age",
+         caption = "Source: FiveThirtyEight") +
+    theme_classic() +
+    theme(plot.title = element_text(hjust = 0.5))
+#print average age of female gun death victims
+average_age <- number_sex_age %>% 
+    filter(sex == 'F') %>% 
+    summarize(mean(age))
+x <- format(average_age, digits = 2)
+print(paste("The average age of female gun death victims is ", x, sep = ""))

```

Comment:
I saw that you get "NaN" as output for x. I think what's happening is that R doesn't recognize your filter(sex == 'F') argument anymore, since you renamed "F" to "Female" above. I think replacing the 'F' with 'Female' should work.

Reply:
Yes, I ignored it. Thanks for pointing out!